src/plugins/validation/2and3/semantic-validators/parameters-ibm.js

```diff
@@ -157,6 +154,36 @@ module.exports.validate = function({ jsSpec, isOAS3 }, config) {
   return { errors: result.error, warnings: result.warning };
 };
 
+function params(path, isOAS3) {
+  const pathsForParameters = [
+    'get',
+    'put',
+    'post',
+    'delete',
+    'options',
+    'head',
+    'patch',
+    'trace',
+    'components'
+  ];
+  const sharedChecks =
+    pathsForParameters.includes(path[path.length - 3]) ||
+    (path[path.length - 4] === 'paths' &&
+      path[2] === 'parameters' &&
+      path.length === 4);
+  if (path[path.length - 2] === 'parameters') {
+    if (isOAS3) {
+      const contentsOfParameterObject = sharedChecks;
+      return contentsOfParameterObject;
+    } else if (!isOAS3) {
+      //swagger 2 allows parameters to be in the top level, this checks for that
+      const contentsOfParameterObject =
+        sharedChecks || (path[0] === 'parameters' && path.length == 2);
+      return contentsOfParameterObject;
+    }
+  }
+}
```

Comment:
This is super difficult to read. I think it would be cleaner to create a variable for each possibility, so it's clear what you are checking for.

Something like this:
```js
const isTopLevelParameter = !isOas3 && path[0] === 'parameters' && path.length === 2;

return isTopLevelParameter || ...
```